includes/rest-api/class-sensei-rest-api-lesson-actions-controller.php

```diff
@@ -0,0 +1,198 @@
+<?php
+/**
+ * File containing the class Sensei_REST_API_Lesson_Actions_Controller.
+ *
+ * @package sensei
+ */
+
+use Sensei\Admin\Content_Duplicators\Lesson_Quiz_Duplicator;
+use Sensei\Admin\Content_Duplicators\Post_Duplicator;
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+/**
+ * Sensei Lesson Actions REST API endpoints.
+ *
+ * @since $$next-version$$
+ */
+class Sensei_REST_API_Lesson_Actions_Controller extends \WP_REST_Controller {
+
+	/**
+	 * Routes namespace.
+	 *
+	 * @var string
+	 */
+	protected $namespace;
+
+	/**
+	 * Routes prefix.
+	 *
+	 * @var string
+	 */
+	protected $rest_base = 'lessons';
+
+	/**
+	 * Cached results of get_item_schema.
+	 *
+	 * @var array
+	 */
+	protected $schema;
+
+	/**
+	 * Sensei_REST_API_Lesson_Actions_Controller constructor.
+	 *
+	 * @param string $namespace Routes namespace.
+	 */
+	public function __construct( $namespace ) {
+		$this->namespace = $namespace;
+	}
+
+	/**
+	 * Register the routes for the controller.
+	 */
+	public function register_routes() {
+		register_rest_route(
+			$this->namespace,
+			$this->rest_base . '/prepare',
```

Comment:
Hum, I think it is better to leave the controller name as is, as we can add other actions here. I didn't find we create a new controller for each endpoint :)

However, I agree, I need to rename the endpoint from `prepare` to `duplicate`. "Prepare" was my previous idea that evolved into duplication :)

Reply:
Hm, not sure if attach is a good option too. Because it's not what it's doing. It's only duplicating the posts and returning the new IDs, right? Maybe it could be a mix of the previous option and your new one? `prepare_to_attach`?

Or being more explicit? `prepare_to_attach_existing_lessons`.

Or something like `maybe_duplicate`? Or is it too bad for an endpoint? haha

Reply:
Yeah, you're right, make sense... I postpone changes here until we decide what to do with duplication. There is a chance we won't need this controller at all if we decide to duplicate lessons on saving the structure.

Reply:
The controller removed entirely.